.pylintrc

```diff
@@ -82,7 +84,6 @@ disable=
  unused-import,
  unused-variable,
  useless-else-on-loop,
- useless-object-inheritance,
```

Comment:
Likewise with this, we need this due to 2.7 as well right?

Reply:
This is about removing the disablement of the check.
@abadger  wrote _"This syntax can also be used in Python 2 if the file specifies __metaclass__ = type at the top of the file."_ which is what he did so enabling this check is fine.

Reply:
If a class inherits we still use object, can that also be safely removed?

Reply:
If I understand what you're asking (what happens when we inherit from an old-style class), then it is not always safe but it is easy to figure out whether we can do it or not.

* In most cases, the class we inherit from probably inherits from object itself.
  * Test this by running `python2.7 -c 'from MODULE import CLASSNAME; print(type(CLASSNAME))'`.  If it prints `<type 'type'>` then it is fine.  If it prints `<type 'classobj'>` then more exploration is needed.
* In other cases, we don't need any features of new-style classes so it does no harm if the class we inherit from is not inheriting from object. (Just two of the features that require a new-style class include using the `@property` decorator and using `super()` inside of a method). If we do need new-style class features then:
* We do need to inherit from both the old-style base class and `object`.  That way our class will be a new-style class and have access to all of the new-style class features.
  * An example is: https://github.com/oamg/convert2rhel/pull/715/files#diff-f82efabf7bb126100afd0472e3d8a5902da6576c6ddf75e3184cce9eb647c52aL218  In Python 2.6, `logging.Formatter` is an old style class and we wantd to use `super()` in our child class.  So we needed to add `object` to the inheritance to make our child class a new-style class.  Note that `logging.Formatter` became a new-style class in Python 2.7 and since we've dropped support for Python-2.6, it is now safe to remove the inheritance from `object`.

Reply:
Gotcha. Thanks for the info!